wordpress-mobile/WordPress-iOS
# History / Revisions: implement local service

Refs. #10304 
Last part fo #10358 

This PR introduces different additions to the History / Revisions task. It includes *WordPressKit 1.4.3-beta.4* and implement the *PostService * extension to handle the GET revisions API.

![revisions](https://user-images.githubusercontent.com/912252/47919144-89a48c80-dea6-11e8-92cb-8ff3fdc06cfb.jpg)

The `PostService+Revisions` extensions calls the remote service:
```
func getPostRevisions(for post: AbstractPost,
                          success: @escaping ([Revision]?) -> Void,
                          failure: @escaping (Error?) -> Void)
```
and map the remote revision into the local managed object model and save the context.

There is a little improvements on the *RevisionsTableViewController* skeleton. Like the posts / pages list, it implements a `WPTableViewHandler` which is the table view datasource, delegate and it handles the fetch result controller.
Also It implements the `RevisionsTableViewPresenter` in charge to call the local service and notify a `RevisionsView` when the call is running or when it's completed.

**To test:**
- Open a post with some revisions, ensure the history button appears and shows a list of revisions
- Revisions should appear on the list
